src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -114,37 +114,87 @@ public class ChangeManagement implements Serializable {
             return returnValue;
 
         }
-
     }
 
     void releaseTransportRequest(String changeId, String transportRequestId, String endpoint, String credentialsId, String clientOpts = '') {
-        int rc = executeWithCredentials( endpoint, credentialsId, 'release-transport', ['-cID', changeId,
-                                                                                        '-tID', transportRequestId], clientOpts) as int
+        int rc = executeWithCredentials( endpoint, credentialsId, 'release-transport', [new KeyValue('cID', changeId),
+                                                                                        new KeyValue('tID', transportRequestId)], clientOpts) as int
         if(rc == 0) {
             return
         } else {
             throw new ChangeManagementException("Cannot release Transport Request '$transportRequestId'. Return code from cmclient: $rc.")
         }
     }
-
+    
     String getCMCommandLine(String endpoint,
                             String username,
                             String password,
                             String command,
-                            List<String> args,
+                            List<CLOption> args,
                             String clientOpts = '') {
         String cmCommandLine = '#!/bin/bash'
         if(clientOpts) {
             cmCommandLine +=  """
                              export CMCLIENT_OPTS="${clientOpts}" """
         }
+        
         cmCommandLine += """
                         cmclient -e '$endpoint' \
                            -u '$username' \
                            -p '$password' \
                            -t SOLMAN \
-                          ${command} ${(args as Iterable).join(' ')}
+                          ${command} ${args.collect{it}.join(' ')}
                     """
+        
         return cmCommandLine
     }
+    
+    abstract static class CLOption 
+    {
+        String key=null
+        String value=null
+        boolean quoteValue=true;
+        String hyphen='-'
+        
+        protected CLOption(String key, String value) {
+            this.key = key
+            this.value = value
+        }
+        CLOption setQuotes(boolean quoteValue) {
+            this.quoteValue=quoteValue
+            return this
+        }
+        CLOption setHyphen(String hyphen) {
```

Comment:
IMO a hyphen is a "-". Naming setHyphen is bogus, since it is possible to set any other value as hyphen. Since it is a string it is also possible to say ```setHyphen("helloWorld")```. 

Is there a use case where we need to set the hyphen to something else like ```-```? If not I would vote for not exposing the possibility to set the "hyphen".

Reply:
i remove the hyphen setter. the only use case was "--return-code".